src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -242,18 +241,26 @@ def _get_docker_command():
             _, stderr = p.communicate()
         except OSError as inner:
             logger.debug("Could not run '%s' command. Exception: %s", docker_command, str(inner))
-            raise CLIError(docker_not_installed)
+            if is_diagnostics_context:
+                return None, DOCKER_COMMAND_ERROR
+            raise CLIError(DOCKER_COMMAND_ERROR.get_error_message())
         except CalledProcessError as inner:
             logger.debug("Could not run '%s' command. Exception: %s", docker_command, str(inner))
-            raise CLIError(docker_not_available)
+            if is_diagnostics_context:
+                return docker_command, DOCKER_DAEMON_ERROR
```

Comment:
I think you can always return None, Error, it also avoids anyone accidentally use "docker" when there is an error

Reply:
The problem with it is that we can run `docker --version` even without docker daemon, so my idea was to return None only if the error was with the command, not with docker daemon. Do you think we should just disregard everything anyways? It might also be a good option...